helix-lock/src/test/java/org/apache/helix/lock/helix/TestZKHelixNonblockingLock.java

```diff
@@ -141,6 +141,23 @@ public void testAcquireLockWhenExistingLockExpired() {
     Assert.assertFalse(_lock.isCurrentOwner());
   }
 
+  @Test
+  public void testOtherClientsAcquireLockFromExistingLockExpired() {
+    // Fake condition when the lock owner is not current user
+    String fakeUserID = UUID.randomUUID().toString();
+    ZNRecord fakeRecord = new ZNRecord(fakeUserID);
+    fakeRecord.setSimpleField(LockInfo.LockInfoAttribute.OWNER.name(), fakeUserID);
+    fakeRecord
+        .setSimpleField(LockInfo.LockInfoAttribute.TIMEOUT.name(), String.valueOf(Long.MAX_VALUE));
+    fakeRecord.setIntField(LockInfo.LockInfoAttribute.PRIORITY.name(), 0);
```

Comment:
I think it's better to also have a test where there is no PRIORITY field set in the ZNRecord, this way the test can verify the case where the lock ZNRecord is written by the old helix-lock lib.

I would also suggest verifying `_lock.tryLock()` returns false instead of throwing exception when there's no PRIORITY field set in the ZNRecord or PRIORITY field is explicitly set to 0 in the ZNRecord, as this was the main compatibility issue that's preventing migration from old to new helix-lock.

Reply:
Thanks. Added a new test down below.